docs/HowtoTDE.md

```diff
@@ -0,0 +1,219 @@
+    vault.hashicorp.com/secret-volume-path-demo: /etc/nuodb/tde/demo
+    vault.hashicorp.com/agent-inject-secret-demo: v2.nuodb.com/TDE/demo
+    vault.hashicorp.com/agent-inject-template-demo: |
+      {{- with secret "v2.nuodb.com/TDE/demo" -}}
+      {{- .Data.data | toJSON }}
+      {{- end -}}
+
+...
+```
+
+Templates for Vault KV store version 1 and version 2 are different..
+If Vault KV store version 1 is used, `.Data.data` references in inject templates should be replaced with `.Data`.
+
+> **NOTE**: For information on how different annotations can be used, see [here](https://www.vaultproject.io/docs/platform/k8s/injector/annotations).
+
+If `tde.conf` needs to be created using several secrets or special keys from the secret needs to be filtered out, [scratch](https://github.com/hashicorp/consul-template#scratch) Consul template API functions can be used. 
+For example this template will filter out any other keys in a Vault secret:
+
+```yaml
+    vault.hashicorp.com/agent-inject-template-demo: |
+      {{- with secret "v2.nuodb.com/TDE/demo" -}}
+      {{- range $k, $v := .Data.data -}}
+      {{- if or (eq $k "target") ($k | regexMatch "historical.*") }}
+      {{- scratch.MapSet "passwords" $k $v }}
+      {{- end }}
+      {{- end }}
+      {{- scratch.Get "passwords" | toJSON }}
+      {{- end -}}
+```
+
+
+## Enable Transparent Data Encryption
+
+NuoDB Admin will configure supplied storage passwords and will propagate them to engines.
+To verify that the target password is configured correctly, the following command should succeed returning no errors.
+
+```bash
+kubectl exec admin-nuodb-cluster0-0 -- nuocmd check data-encryption \
+  --db-name demo \
+  --password 'topSecret'
+```
+
+Now that the database has a storage password, encryption can be enabled using `ALTER DATABASE CHANGE ENCRYPTION TYPE`. Database administrator privileges are required in order to use this command.
+The SQL statement should be executed manually by obtaining SQL connection towards the target database.
+
+```bash
+kubectl exec -n nuodb admin-nuodb-cluster0-0 -- bash -c \
+  'echo "alter database change encryption type AES128;" | /opt/nuodb/bin/nuosql demo --user dba --password secret'
+```
+
+SMs will start data encryption in the background.
+
+> **NOTE**: For information on how to confirm if TDE is enabled, see [here](https://doc.nuodb.com/nuodb/latest/database-administration/transparent-data-encryption/configuring-transparent-data-encryption/#_confirming_that_tde_is_enabled).
+
+
+## Change storage passwords
+
+Database storage passwords can be changed at any time by changing the secret supplying them. The password rotation will be performed automatically in the background.
+
+The current storage password must be specified as historical so that it can be used if:
+
+- some of the SMs are down during password rotation
+- database restore needs to be performed from a backupset encrypted with the previous password
```

Comment:
I know what this paragraph is saying, but it could use some rewording. 

```
Historical passwords are not required during normal operation. There are special circumstances under which a historical password needs to be provided. These circumstances include:
- some of the Storage Managers were down during password rotation. Once these SMs start, they will need to use the historical password to read the archive so they can encrypt it with the new target password.
- A DBA wants to use an older backupset that was encrypted with a previous historical password. In order for an SM to decrypt an older backupset, it must be provided a historical password that matches the password that was used to encrypt the backupset.
```

Reply:
the second bullet might be an overkill. You already explain it well at the end of the doc. Maybe we just need to refer to that section here.